weave/flow/eval.py

```diff
@@ -268,6 +280,121 @@ async def evaluate(self, model: Union[Op, Model]) -> dict:
 
         return summary
 
+    def get_evaluate_calls(self) -> CallsIter:
+        """
+        Retrieve all evaluation calls that used this Evaluation object.
+
+        Note that this returns a CallsIter instead of a single call because it's
+        possible to have multiple evaluation calls for a single evaluation (e.g.
+        if you run the same evaluation multiple times).
+
+        Returns:
+            CallsIter: An iterator over Call objects representing evaluation runs.
+
+        Raises:
+            ValueError: If the evaluation has no ref (hasn't been saved/run yet).
+
+        Examples:
+            >>> evaluation = Evaluation(dataset=examples, scorers=[scorer])
+            >>> await evaluation.evaluate(model)  # Run evaluation first
+            >>> calls = evaluation.get_evaluate_calls()
+            >>> for call in calls:
+            ...     print(f"Evaluation run: {call.id} at {call.started_at}")
+        """
+        evaluate_op_name = "Evaluation.evaluate"
+
+        if not self.ref:
+            raise ValueError("Evaluation has no ref, please run the evaluation first!")
+
+        client = require_weave_client()
+        return client.get_calls(
+            query={
+                "$expr": {
+                    "$contains": {
+                        "input": {"$getField": "op_name"},
+                        "substr": {"$literal": evaluate_op_name},
+                    }
+                }
+            },
+            filter=CallsFilter(input_refs=[self.ref.uri()]),
+        )
+
+    def get_score_calls(self) -> dict[str, list[Call]]:
```

Comment:
this is a bit surprising that calling this one method gets **all** scores for **all** evaluations. Not sure when I would ever call this. I think I would more frequently use the get all scores for a **given** evaluation (pass in a call or call_id?).

Reply:
I thought about this, but I don't think there's an easy way for a user to know the exact trace_id of the a given evaluation.

Reply:
I find myself doing a get all root traces query much of the time, that would end up in a list of evals with trace_ids, then i might want scores for the last 5 of them or whatever

Reply:
Maybe this should expect a run as input, i.e. an `evaluate` call.

Longer term, should we add a `Run` class, that would essentially wrap around `Evaluation.evaluate` calls? Then we could define APIs such as `evaluation.get_runs()` and `run.get_scores()`.

Could look like
```python
class Run:
    id: str # root_call.trace_id
    root_call: Call # The Evaluation.evaluate call
```